custom_components/adaptive_lighting/switch.py

```diff
@@ -238,57 +237,136 @@ def _split_service_data(service_data, adapt_brightness, adapt_color):
     return service_datas
 
 
-async def handle_apply(switch: AdaptiveSwitch, service_call: ServiceCall):
-    """Handle the entity service apply."""
-    hass = switch.hass
-    data = service_call.data
-    all_lights = data[CONF_LIGHTS]
-    if not all_lights:
-        all_lights = switch._lights
-    all_lights = _expand_light_groups(hass, all_lights)
-    switch.turn_on_off_listener.lights.update(all_lights)
+def _parse_service_args(hass: HomeAssistant, service_call: ServiceCall):
     _LOGGER.debug(
-        "Called 'adaptive_lighting.apply' service with '%s'",
-        data,
+        "Called '_parse_service_args' with service data:\n'%s'",
+        service_call.data,
     )
-    for light in all_lights:
-        if data[CONF_TURN_ON_LIGHTS] or is_on(hass, light):
-            await switch._adapt_light(  # pylint: disable=protected-access
-                light,
-                data[CONF_TRANSITION],
-                data[ATTR_ADAPT_BRIGHTNESS],
-                data[ATTR_ADAPT_COLOR],
-                data[CONF_PREFER_RGB_COLOR],
-                force=True,
-                context=switch.create_context("service", parent=service_call.context),
+    data = service_call.data
+    lights = data[CONF_LIGHTS]
+    switch_unique_id = data.get("entity_id")
```

Comment:
Is `switch_unique_id` a list of IDs here?

Reply:
yes it is a list: a list of switch unique ids. Validates with `cv.entity_ids`.

During testing, I believe `switch_unique_id` started off as a switch object, then a string, then became a dictionary structured as `{<switch unique id>: <switch object>}`, then ended up as a list. That variable has had quite the adventure lol.